torch/nn/modules/pooling.py

```diff
@@ -236,7 +236,7 @@ class MaxPool3d(_MaxPoolNd):
     padding: _size_3_t
     dilation: _size_3_t
 
-    def forward(self, input: Tensor) -> Tensor:
+    def forward(self, input: Tensor):
```

Comment:
Are we actually ok to regress on these type annotations? Or are they moved somewhere else?

Reply:
these type annotations weren't correct

Reply:
Should we have it be `Union[Tensor, Tuple[Tensor, Tensor]]` then?
My guess is that this wasn't a union before as jit didn't support it?

Reply:
@albanD see the PR description for commentary on `Union[Tensor, Tuple[Tensor, Tensor]]`. I can go that route if you're OK with me inserting lots of `assert isinstance(output, Tensor)` throughout the code base (including in torchvision and potentially other user code).

Reply:
You would need these to appease mypy?

Reply:
We would want to type these as `Tensor` or `Tensor, Tensor` not the Union of them. See:  https://github.com/pytorch/pytorch/issues/39670. However IMO I don't think that should block this PR, as is the type annotation is wrong.

Reply:
Interesting issue, thanks for the context!
I guess this is ok yes.

I was wondering if we could trick the system by doing the asserts inside the pooling function, based on the given flag. But I guess that the issue answers the question: ts doesn't do the type inference at all so that won't...

Reply:
Thanks for the helpful conversation. Marking this resolved.